source/Hardpoint.cpp

```diff
@@ -201,36 +183,53 @@ void Hardpoint::Step()
 	// If the full reload time is elapsed, reset the burst counter.
 	if(reload <= 0.)
 		burstCount = outfit->BurstCount();
+	// If the weapon recently fired then increment any sustained fire counters.
+	if(reload > 0.)
+	{
+		if(bloomCount < outfit->BloomTime())
+			++bloomCount;
+		if(spinupCount < outfit->SpinupTime())
+			++spinupCount;
+	}
+	// Only decrement sustained fire counters if this weapon could have fired
+	// but didn't.
+	else if(!wasFiring)
+	{
+		if(bloomCount > 0.)
+			--bloomCount;
+		if(spinupCount > 0.)
+			--spinupCount;
+	}
+
 	if(burstReload > 0.)
 		--burstReload;
 	// If the burst reload time has elapsed, this weapon will not count as firing
 	// continuously if it is not fired this frame.
-	if(burstReload <= 0. && reload <= 0)
+	if(burstReload <= 0.)
 		isFiring = false;
-	if(outfit->BurstCount() > 1 && burstCount == 0)
+	// If a burst has run out then reset the burst sustained fire counters.
+	if(burstCount == 0)
 	{
-		burstSpinupCount = outfit->BurstSpinupTime();
-		if(wasFiring && spinupCount > 0)
-			--spinupCount;
-		else if(!wasFiring && spinupCount < outfit->SpinupTime())
-			++spinupCount;
+		burstBloomCount = 0.;
+		burstSpinupCount = 0.;
 	}
-	else
+	// If the weapon recently fired a burst then increment any burst
+	// sustained fire counters.
+	else if(burstReload > 0.)
 	{
```

Comment:
Currently burst weapons with burst reload of 1 (Gatling Gun is what I tested) don't get bloom.inaccuracy added.  This just moves the burstReload decrement after the check for burstBloomCount incrementing.  And yes...it'll throw off the great big suggestion above this...I don't know if you can update suggestions but I'll try.
```suggestion
	// If the burst reload time has elapsed, this weapon will not count as firing
	// continuously if it is not fired this frame.
	if(burstReload <= 0.)
		isFiring = false;
	// If a burst has run out then reset the burst sustained fire counters.
	if(burstCount == 0)
	{
		burstBloomCount = 0.;
		burstSpinupCount = 0.;
	}
	// If the weapon recently fired a burst then increment any burst
	// sustained fire counters.
	else if(burstReload > 0.)
	{
			--burstReload;
```